src/App.tsx

```diff
@@ -1,27 +1,92 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
+import Button from '@mui/material/Button';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [selectButton, setSelectButton] = useState('');
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const handleClickAll = () => {
+    getAll()
+      .then((good: Good[]) => {
+        if (selectButton !== 'all') {
+          setSelectButton('all');
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+          return setGoods(good);
+        }
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+        setSelectButton('');
 
-    <GoodsList goods={[]} />
-  </div>
-);
+        return setGoods([]);
+      });
+  };
+
+  const handleClick5First = () => {
+    get5First()
+      .then((good: Good[]) => {
+        if (selectButton !== 'five') {
```

Comment:
```suggestion
        if (selectButton !== 'five') {
```

lets create a enum for 'five', 'red' and 'all'

Reply:
I change all comments, but i dont understand how to use enum here(

Reply:
😱